textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/RegexPredicate.java

```diff
@@ -36,5 +50,27 @@ public boolean satisfy(ITuple tuple) {
         }
         return false;
     }
+    
+    public List<Span> statisfySpan(ITuple tuple) {
+    	List<Span> res = new ArrayList<>();
+    	if (tuple == null) {
+    		return res; //empty array
+    	}
+    	IField field = tuple.getField(fieldName);

```

Comment:
@laisycs @chenlica @sandeepreddy602 
We should notice the fundamental difference between regex matching and dictionary/keyword matching. While tokenization (removing spaces, punctuations, stop words) helps for dictionary matching, it cannot be used in regex, because the point of regex is matching **exactly** based on the regex. 
The last step of regex matching is always scan based, no matter how Lucene internally treats TextField and StringField differently, they should be treated the same in regex matching.
So I suggest to modify it to 
`if (field instanceof StringField || field instanceof TextField)`